### PR TITLE
Use GetExportedTypes where possible and filter nested types

### PR DESCRIPTION
**Pull request by filmor**
This is basically what @dmitriyse prepared in PR 528 without the event code that I don't understand.

### What does this implement/fix? Explain your changes.

Useless calls to load dependencies based on private and nested types.

### Does this close any currently open issues?

#703 

### Any other comments?

...

### Checklist

Check all those that are applicable and complete.

-   [ ] Make sure to include one or more tests for your change
-   [x] If an enhancement PR, please create docs and at best an example
-   [x] Add yourself to [`AUTHORS`](../blob/master/AUTHORS.md)
-   [x] Updated the [`CHANGELOG`](../blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by filmor** [comment]
@denfromufa Any idea for a test?

**Comment by denfromufa** [comment]
@filmor i don't have a good option for a test (requires multiple assemblies), unless we ask to test out this PR from all users @goerch @pierce-martin @williamrubens @inna-w @Cronan @barrybriggs @ZhukovGreen @AlexCatarino who reported this in the bug tracker:

#703 
#643
#622 
#603 
#598 
#528 
https://github.com/MSLNZ/msl-loadlib/issues/6

**Comment by denfromufa** [comment]
@filmor i think this PR still misses this essential part about handling exceptions from "transitive" dependencies, which I experienced before (I'm going to find a link), see this comment from @Cronan:

https://github.com/pythonnet/pythonnet/pull/528/files#r167013554

File: src/runtime/assemblymanager.cs [hunk 1]
```diff
@@ -461,5 +458,13 @@ public static Type LookupType(string qname)
             }
             return null;
         }
+
+        internal static Type[] GetTypes(Assembly a)
```
Comment (denfromufa) [inline src/runtime/assemblymanager.cs]: need a try-catch block here for transitive depedencies

**Comment by denfromufa** [comment]
@filmor i agree with your comment about skiploadassembly: https://github.com/pythonnet/pythonnet/pull/528#issuecomment-414668945

File: src/runtime/assemblymanager.cs (hunk 1 above)
Comment (filmor) [reply src/runtime/assemblymanager.cs]: Isn't it an error, though, if we can't load transitive dependencies?

**Comment by filmor** [comment]
@denfromufa Can you please provide me with a test-case for the transitive dependency issue you mention? I have really no idea what this is about.

File: src/runtime/assemblymanager.cs [hunk 2]
```diff
@@ -470,9 +471,28 @@ public static Type LookupType(string qname)
...
+                try
+                {
+                    return a.GetTypes();
+                }
+                catch (ReflectionTypeLoadException exc)
+                {
+                    // Return all types that were successfully loaded
+                    return exc.Types.Where(x => x != null).ToArray();
+                }
+            }
```
Comment (filmor) [inline src/runtime/assemblymanager.cs]: @denfromufa Is this what you had in mind?

Comment (denfromufa) [reply src/runtime/assemblymanager.cs]: @filmor this is even more than what i asked for :)

**Pull request merged**